src/reanimated2/shareables.ts

```diff
@@ -149,6 +149,12 @@ export function makeShareableCloneRecursive<T>(
               value.__stackDetails
             );
             delete value.__stackDetails;
+          } else if (value.__stackDetails) {
+            // Detected debug version of the worklet in release bundle. This
+            // might lead to unexpected issues or errors. Probably one of user
+            // dependencies provided transpiled code with debug version of the
+            // Reanimated plugin.
+            delete value.__stackDetails;
```

Comment:
Let's throw an error here to indicate that this configuration is not supported.
```suggestion
            throw new Error('[Reanimated] Using dev bundle in a release app build is not supported');
```

Reply:
My app is crashing in production too due this issue https://github.com/software-mansion/react-native-reanimated/issues/4690. Is this error throw really necessary? Guess it can crash app again.

Reply:
We cannot guarantee that the app will work fine in release mode when using dev bundle so I guess it's better to fail immediately.

Reply:
But there is any cases of release mode using dev bundle? How could I reproduce this scenario?

Reply:
It seems like https://github.com/software-mansion/react-native-reanimated/issues/4690#issuecomment-1634337256 and https://github.com/software-mansion/react-native-reanimated/issues/4690#issuecomment-1634344638 should answer your question.